src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -159,9 +166,14 @@ public async Task<FormulaValue> EvalAsync(string expressionText, CancellationTok
 
             var check = Check(expressionText, (RecordType)parameters.IRContext.ResultType, options);
             check.ThrowOnErrors();
-
-            var newValue = await check.Expression.EvalAsync(parameters, cancel);
-            return newValue;
+            try
+            {
+                return await check.Expression.EvalAsync(parameters, cancel);
+            }
+            catch (MaxCallDepthException)

```

Comment:
>catch (MaxCallDepthException) [](http://example.com/codeflow?start=12&length=29)

This should be deeper - in the ParsedExpression.EvalAsync  call directly.  There are ways to call ParsedExpression that don't go through Engine.EvalAsync.  (such as getting it from Check).